tests/unit/preprocessor/KernelPCA_unittest.cc

```diff
@@ -7,37 +7,63 @@
 using ::testing::Test;
 using namespace shogun;
 
-#ifdef HAVE_LAPACK
-TEST(KernelPCA, DISABLED_apply_to_feature_matrix_input)
+TEST(KernelPCA, apply_to_feature_matrix_input)
 {
-	float64_t data[] = {1, 1, 1,
-                      1, 2, 3,
-                      5, 6, 1,
-                      2, 2, 2,
-                      1, 1, 1};
-	float64_t resdata[] = {-1.526879008202007e-02,  6.902776989923266e-01,
-                         -4.032822763552926e-01, -5.151523890814317e-01,
-                         8.444041004961732e-01, -4.318711485273607e-01,
-                         -4.105842439768400e-01, -4.335318603758601e-01,
-                         -1.526879008202015e-02, 6.902776989923268e-01
-	                        };// column-wise
+	float64_t data[] = {1, 1, 1, 1, 2, 3, 5, 6, 1, 2, 2, 2, 1, 1, 1};
+	float64_t test_data[] = {3, 3, 3, 7, 4, 1};
+	float64_t resdata[] = {-0.17645841, 0.013962, -0.16082441, 0.03640145};
+
 	int32_t num_vectors = 5;
 	int32_t num_features = 3;
-	SGMatrix<float64_t> orig(data, num_features, num_vectors, false);
-	SGMatrix<float64_t> m = orig.clone();
+	int32_t num_test_vectors = 2;
+	int32_t target_dim = 2;
+
+	SGMatrix<float64_t> m(data, num_features, num_vectors, false);
 	CDenseFeatures<float64_t>* feats = new CDenseFeatures<float64_t>(m);
+
 	CGaussianKernel* kernel = new CGaussianKernel();
 	kernel->set_width(1);
+
 	CKernelPCA kpca(kernel);
-	kpca.set_target_dim(2);
+	kpca.set_target_dim(target_dim);
 	kpca.init(feats);
-	SGMatrix<float64_t> embedding = kpca.apply_to_feature_matrix(feats);
 
-	float64_t s;
+	SGMatrix<float64_t> test_matrix(
+	    test_data, num_features, num_test_vectors, false);
+	CDenseFeatures<float64_t>* feats_test =
+	    new CDenseFeatures<float64_t>(test_matrix);
+	SGMatrix<float64_t> embedding = kpca.apply_to_feature_matrix(feats_test);
+
+	// allow embedding with opposite sign
+	for (index_t i = 0; i < target_dim * num_test_vectors; ++i)
+		EXPECT_LE(
+		    CMath::abs(embedding.matrix[i]) - CMath::abs(resdata[i]), 1E-6);
+}
+
+TEST(KernelPCA, apply_to_feature_vector_input)
+{
+	float64_t data[] = {1, 1, 1, 1, 2, 3, 5, 6, 1, 2, 2, 2, 1, 1, 1};
```

Comment:
I have a gist somewhere...here we go https://gist.github.com/micmn/f93f723b74db2a1eb5875f63d841bdc1